drivers/sensor/icm42605/icm42605_trigger.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2016 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <sys/util.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+#include "icm42605.h"
+#include "icm42605_setup.h"
+
+LOG_MODULE_DECLARE(ICM42605, CONFIG_SENSOR_LOG_LEVEL);
+
+int icm42605_trigger_set(struct device *dev,
+			 const struct sensor_trigger *trig,
+			 sensor_trigger_handler_t handler)
+{
+	struct icm42605_data *drv_data = dev->data;
+	const struct icm42605_config *cfg = dev->config;
+
+	if (trig->type != SENSOR_TRIG_DATA_READY && trig->type != SENSOR_TRIG_TAP && trig->type != SENSOR_TRIG_DOUBLE_TAP) {
+		return -ENOTSUP;
+	}
+
+	gpio_pin_interrupt_configure(drv_data->gpio, cfg->int_pin,
+				     GPIO_INT_DISABLE);
+
+	if (handler == NULL) {
+		icm42605_turn_off_sensor(dev);
+		return 0;
+	}
+
+	if (trig->type == SENSOR_TRIG_DATA_READY) {
+		drv_data->data_ready_handler = handler;
+		drv_data->data_ready_trigger = *trig;
+	} else if (trig->type == SENSOR_TRIG_TAP || trig->type == SENSOR_TRIG_DOUBLE_TAP) {
+		drv_data->tap_handler = handler;
+		drv_data->tap_trigger = *trig;
+	} else {
+		return -ENOTSUP;
+	}
+
+	gpio_pin_interrupt_configure(drv_data->gpio, cfg->int_pin,
+				     GPIO_INT_EDGE_TO_ACTIVE);
+
+	k_msleep(100);
```

Comment:
Sleep time should be defined and documented.

Reply:
I will document soon.